src/main/java/org/apache/commons/imaging/common/RationalNumber.java

```diff
@@ -73,8 +124,41 @@ private static long gcd(final long a, final long b) {
         return gcd(b, a % b);
     }
 
+    /**
+     * Negates the value of the RationalNumber. If the numerator of this
+     * instance has its high-order bit set, then its value is too large
+     * to be treated as a Java 32-bit signed integer.  In such a case, the
+     * only way that a RationalNumber instance can be negated is to
+     * sacrifice one bit of precision by dividing both the numerator and
+     * the denominator by 2. However, if that computation would result
```

Comment:
by `2` or by the greatest common divisor?

Reply:
by the computed greatest common denominator.

Reply:
done